### ISSUE TITLE
Assertion violation in DQMServices/Core/interface/DQMStore.h

### ISSUE THREAD

**Issue by silviodonato** [cms-sw/cmssw#32563]
About one hundred of workflows are crashing in `CMSSW_11_3_X_2020-12-18-2300` with
```
cmsRun: /data/cmsbld/jenkins/workspace/build-any-ib/w/slc7_amd64_gcc900/cms/cmssw/CMSSW_11_3_X_2020-12-17-2300/src/DQMServices/Core/interface/DQMStore.h:611: dqm::implementation::DQMStore::bookTransaction(iFunc, uint64_t, bool)::ModuleIdScope::ModuleIdScope(dqm::implementation::IBooker&, uint64_t, MonitorElementData::Scope, edm::LuminosityBlockID) [with iFunc = DQMOneEDAnalyzer<Args>::beginRun(const edm::Run&, const edm::EventSetup&) [with Args = {}]::<lambda(dqm::legacy::DQMStore::IBooker&)>; uint64_t = long unsigned int]: Assertion `oldid_ == 0 || !"Nested booking transaction?"' failed.
```
see https://cmssdt.cern.ch/SDT/html/cmssdt-ib/#/relVal/CMSSW_11_3/2020-12-18-2300?selectedArchs=slc7_amd64_gcc900&selectedFlavors=X&selectedStatus=failed

In CMSSW_11_3_X_2020-12-18-2300 we merged few PRs ( #32510, #32535, #32537 ) so I suspect the origin of the problem is #32510 (I'm running a test to get confirm)

**Comment by silviodonato** [comment]
assign dqm, core

**Comment by Dr15Jones** [comment]
@silviodonato I took a look at the stack trace and I agree it almost certainly comes from #32510. I saw that a request to get data from the EventSetup ultimately caused a EDProducer to be run on that waiting thread. The use of a `tbb::task_arena` for all EventSetup calls was supposed to prevent such behavior. It looks like that recent pull request accidently caused the `tbb::task_arena` to be bypassed in some cases.

I'll take a look at it today to see if I can quickly fix the problem.

**Comment by silviodonato** [comment]
Thanks @Dr15Jones. I tried to reproduce the error using `CMSSW_11_3_X_2020-12-18-1100` + #32510, but the compilation takes ages. I've tried the other way around using `CMSSW_11_3_X_2020-12-18-1100` + #32535, #32537 and it works. In `CMSSW_11_3_X_2020-12-18-2300`, I see that the error appears only using `-t 4` in `runTheMatrix`.

**Comment by Dr15Jones** [comment]
@silviodonato I see exactly where I made the mistake. I'll have a fix very soon. The hardest part is trying to reproduce the failure :).

**Comment by Dr15Jones** [comment]
Should be fixed in #32566. My attempts to reproduce the problem didn't succeed but I'll try running one of the rel vals over and over to see if the problem returns after this change.

**Comment by silviodonato** [comment]
I reproduced the error with `runTheMatrix.py -l 4.51 -i all --job-reports -t 4 --ibeos`

**Comment by silviodonato** [comment]
fixed by #32566
The fix works, see
https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_amd64_gcc900/CMSSW_11_3_X_2020-12-20-0000/pyRelValMatrixLogs/run/110.0_SingleTauPt50Pythia+SingleTauPt50Pythia+DIGI+RECO/step3_SingleTauPt50Pythia+SingleTauPt50Pythia+DIGI+RECO.log#/
vs
https://cmssdt.cern.ch/SDT/cgi-bin/logreader/slc7_amd64_gcc900/CMSSW_11_3_X_2020-12-19-1100/pyRelValMatrixLogs/run/110.0_SingleTauPt50Pythia+SingleTauPt50Pythia+DIGI+RECO/step3_SingleTauPt50Pythia+SingleTauPt50Pythia+DIGI+RECO.log#/
